### PR TITLE
Versioned API

### PR DESCRIPTION
**Pull request by lorden**
Currently API versions are not supported in swagger, but that's a key part of any API. 
The goal is to have versions as a completely optional feature, so I added a few changes to allow documentation for different API versions:

- Added versioned URLs to allow passing a specific version `.../docs/v1/` or `.../docs/v2/`
- The views will read this param and set it to `SWAGGER_SETTINGS['api_version']`, with this value your app can do whatever it needs to show the right version.
- Added `available_api_versions` to settings to have a limited set of versions available, this way we can deprecate old versions.
- Added versioning usage to the docs and tests

### CODE REVIEW

**Comment by linuxlewis** [comment]
:+1:

**Comment by ariovistus** [comment]
I'm trying to get a copy of cigar_example working with versioning. I changed the urls like so: ```r'^v2/drop_cigar_in_jambalaya/$'```, I changed SWAGGER_SETTINGS to have 
```
    'api_version': 'v2', 
    'available_api_versions': ['v2'],
```

and it gives me

NoReverseMatch at /

Reverse for 'django.swagger.base.view' with arguments '('v2',)' and keyword arguments '{}' not found. 0 pattern(s) tried: []

what am I doing wrong?

**Comment by lorden** [comment]
I think you're mixing versions in cigar with versions in swagger. For swagger the configuration is almost right, drop the "v"s in api_version and available_api_version, so:
```
'api_version': '2',
'available_api_versions': ['2']
```
Then you can go to your docs with [...]/v2/, that will set "2" as the `SWAGGER_SETTINGS['api_version']`. From there what the project that is isolated from swagger, what you get is that number set in the settings so you can respond with the right content. 
Now you made me realize that an example project would be very helpful to see how versions can be implemented. I'll try to add that here.

**Comment by omatvejev** [comment]
Has there been any progress on versioning support?

**Comment by asergeant01** [comment]
This seems exactly what I'm after as well..... any update?

**Comment by CBurini** [comment]
I'm also very (very) interested in this feature ! I would love for it to be integrated into the project.

**Comment by adrienrusso** [comment]
:+1:

File: rest_framework_swagger/views.py [hunk 1]
```diff
@@ -41,7 +44,12 @@ def get_restructuredtext(view_cls, html=False):
 
 class SwaggerUIView(View):
 
-    def get(self, request, *args, **kwargs):
+    def get(self, request, api_version=None, *args, **kwargs):
+
+        if api_version not in SWAGGER_SETTINGS.get('available_api_versions') and SWAGGER_SETTINGS['api_version']:
+            return redirect(reverse('django.swagger.base.view', args=(SWAGGER_SETTINGS['api_version'],)))
```
Comment (blueyed) [inline rest_framework_swagger/views.py]: This should maybe raise an error instead?

**Comment by blueyed** [comment]
Would be really useful.
Additionally (or maybe I've just missed it), it would be great to have a dropdown for the available versions (but that's only nice-to-have for now).

A related PR to handle the version in the request header: https://github.com/marcgibbons/django-rest-swagger/pull/379.

**Comment by lorden** [comment]
Not planning to update this, closing it.

**Pull request closed without merging**